applications/vanilla/controllers/class.vanillasettingscontroller.php

```diff
@@ -460,6 +460,10 @@ public function deleteCategory($CategoryID = false) {
         // Get category data
         $this->Category = $this->CategoryModel->getID($CategoryID);
 
+        // Block deleting special categories.
+        if (val('CanDelete', $this->Category) === 0) {
```

Comment:
I'd do:

```php
if (val('CanDelete', $this->Category, true)) {
```

You want this to be resilient for the case where the database isn't updated. I also think the database brings the value in as a string doesn't it?